src/app/directives/drag-and-drop.directive.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Directive,
+  HostBinding,
+  HostListener,
+  Output,
+  EventEmitter
+} from "@angular/core";
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+
+export interface FileHandle {
+  file: File,
+  url: SafeUrl
+}
+
+@Directive({
+  selector: '[appDragAndDrop]'
+})
+export class DragAndDropDirective {
+  @Output() files: EventEmitter<FileHandle[]> = new EventEmitter();
+
+  constructor(private sanitizer: DomSanitizer) { }
+
+  @HostBinding("style.opacity") private opacity = '1';
+
+  @HostListener("dragover", ["$event"]) public onDragOver(evt: DragEvent) {
+    this.opacity = '0.7';
+    evt.preventDefault();
+    evt.stopPropagation();
+  }
+
+  @HostListener("dragleave", ["$event"]) public onDragLeave(evt: DragEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+  }
+
+  @HostListener('drop', ['$event']) public onDrop(evt: DragEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+    
+    let files: FileHandle[] = [];
+    for (let i = 0; i < evt.dataTransfer.files.length; i++) {
```

Comment:
Please rewrite it using 'forEach'

Reply:
It is better to use for loop than forEach for FileList
source: https://stackoverflow.com/questions/50604272/typescript-property-foreach-does-not-exist-on-type-filelist